### PR TITLE
add batch_norm op with test and benchmark

### PR DESCRIPTION
**Pull request by yanghailong-git**
## Summary

Implemented a 2D batch normalization Triton operator, successfully ran the corresponding tests and benchmarks, and visualized the performance tests for speed and memory.

## Testing Done

- Hardware Type: <BLANK>
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

the visualization of performance:
batch_norm_speed
batch_norm_memory

### CODE REVIEW

**Comment by yundai424** [comment]
looks like from the benchmark result triton impl is slower than HF original one? 👀

**Comment by yanghailong-git** [comment]
> looks like from the benchmark result triton impl is slower than HF original one? 👀

It seems so. The memory usage is about the same, but the speed is a bit slower. Do you have any optimization or improvement methods?
